src/Hosting/Hosting/src/Internal/HostingEventSource.cs

```diff
@@ -19,6 +19,10 @@ internal sealed class HostingEventSource : EventSource
     private long _currentRequests;
     private long _failedRequests;
 
+    public long TotalRequests => Volatile.Read(ref _totalRequests);
+    public long CurrentRequests => Volatile.Read(ref _currentRequests);
+    public long FailedRequests => Volatile.Read(ref _failedRequests);
```

Comment:
There is no mention of dimensions here, which is half the value for Metrics. 
At the Host level, there possibly are not many applicable dimensions. You typically want to keep the number of dimensions small (handful at most) and not many values per dimension.
FailedRequests may be applicable to have a status code for the dimension - eg 404, 500 etc.

Reply:
I've added status code to request end